models/vision/detection/yolo/yolov3_postproc_ipu.cc

```diff
@@ -0,0 +1,217 @@
+#include "yolo_anchors.txt"
+};
+
+template <int batch, int dim, int cls_num>
+static void decode(int orig_img_w, int orig_img_h, float* bb,
+                   const std::vector<int>& anchor_mask) {
+  int num_anchors = anchor_mask.size();
+  float scale = std::min(416.0 / orig_img_h, 416.0 / orig_img_w);
+  float new_shape_h = std::round(orig_img_h * scale);
+  float new_shape_w = std::round(orig_img_w * scale);
+  float offset_h = (416 - new_shape_h) / 2.0 / 416;
+  float offset_w = (416 - new_shape_w) / 2.0 / 416;
+  float scale_h = 416 / new_shape_h;
+  float scale_w = 416 / new_shape_w;
+
+  assert(num_anchors == 3);
+  std::vector<std::pair<int, int>> anchors;
+  for (auto m : anchor_mask) {
+    anchors.push_back({AllAnchors[m * 2], AllAnchors[m * 2 + 1]});
+  }
+// [batch, dim, dim, num_anchors, 5 + cls_num]
+// dx/dy [:, :, :, :, 0:2]
+// dw/dh [:, :, :, :, 2:4]
+// conf: [:, :, :, :, 4:5]
+// prob: [:, :, :, :, 5: ]
+#pragma omp parallel for
+  for (size_t i = 0, e = batch * dim * dim * num_anchors * (cls_num + 5);
+       i != e; ++i) {
+    bb[i] = std::exp(bb[i]);
+  }
+  float* ptr = bb;
+  for (int i = 0; i != batch; ++i) {
+    for (int grid_y = 0; grid_y < dim; ++grid_y) {
+      for (int grid_x = 0; grid_x < dim; ++grid_x) {
+        for (int a = 0; a < num_anchors; ++a) {
+          auto dx = ((ptr[0] / (ptr[0] + 1)) + grid_x) / dim; // dx
+          dx = (dx - offset_w) * scale_w;
+          float dy = ((ptr[1] / (ptr[1] + 1)) + grid_y) / dim; // dy
+          dy = (dy - offset_h) * scale_h;
+
+          float dw = (ptr[2] * anchors[a].first) / 416 * scale_w;
+          float dh = (ptr[3] * anchors[a].second) / 416 * scale_h;
+
+          *ptr++ = (dy - dh / 2.0) * orig_img_h; // y_min
+          *ptr++ = (dx - dw / 2.0) * orig_img_w; // x min
+          *ptr++ = (dy + dh / 2.0) * orig_img_h; // y_max
+          *ptr++ = (dx + dw / 2.0) * orig_img_w; // x max
+
+          float confidence = (*ptr) / ((*ptr) + 1);
+          *ptr++ = confidence;
+
+          for (int c = 0; c < cls_num; ++c, ++ptr) {
+            *ptr = (*ptr) / ((*ptr) + 1) * confidence;
+          }
+        }
+      }
+    }
+  }
+}
+// Using Custom Op. Too Slow...
+std::vector<std::pair<std::string, std::array<float, 5>>> post_process_nhwc(
+    int orig_img_w, int orig_img_h, float bb13[1 * 13 * 13 * 255],
+    float bb26[1 * 26 * 26 * 255], float bb52[1 * 52 * 52 * 255]) {
+  assert(ClassesNames.size() == 80);
+  const std::vector<std::vector<int>> anchor_masks{
+      {6, 7, 8}, {3, 4, 5}, {0, 1, 2}};
+  decode<1, 13, 80>(orig_img_w, orig_img_h, bb13, anchor_masks[0]);
+  decode<1, 26, 80>(orig_img_w, orig_img_h, bb26, anchor_masks[1]);
+  decode<1, 52, 80>(orig_img_w, orig_img_h, bb52, anchor_masks[2]);
+  // NMS
+  float score_thre = 0.3;
+  float iou_thre = 0.45;
+  std::vector<std::pair<std::string, std::array<float, 5>>> ret;
+
+  void* handle =
+      dlopen("../ODLA/platforms/odla_popart/custom_ops/build/libcustom_ops.so",
```

Comment:
can we use libODLA_popart? In yolov3 code, we don't use IPU APIs directly. Instead, we ODLA apis.
We can construct a odla_computation and use odla_NMS() here.